### PR TITLE
Update Travis CI script

### PR DESCRIPTION
**Pull request by mforets**
EDIT:

Some relevant links:

- https://gist.github.com/jamesarosen/e29076bd81a099f0f72e
- https://github.com/phpbb/phpbb/blob/3.1.x/.travis.yml
- https://github.com/pantheon-deprecated/travis-scripts/blob/master/examples/example.travis.yml

### CODE REVIEW

**Comment by mforets** [comment]
That's the kind of problem addressed and solved by  [Pkg3](https://docs.julialang.org/en/v1/manual/code-loading/#Federation-of-packages-1) -- I've opened https://github.com/JuliaReach/Reachability.jl/issues/600 as follow up.

File: .travis.yml [hunk 1]
```diff
@@ -12,19 +12,20 @@ git:
...
 script:
- - julia -e 'using Pkg;
-             Pkg.develop("LazySets");
-             Pkg.develop("MathematicalSystems");
-             Pkg.develop("HybridSystems");
-             Pkg.develop("TaylorModels");
-             Pkg.clone(pwd());
-             Pkg.build("Reachability");
-             Pkg.test("Reachability"; coverage=true)'
+ - julia -e $TEST_SUITE
```
Comment (schillic) [inline .travis.yml]:
```suggestion
 - julia -e '$TEST_SUITE'
```

Comment (schillic) [reply .travis.yml]: That did not help :disappointed:

File: .travis.yml [hunk 2]
```diff
@@ -12,19 +12,20 @@ git:
...
 script:
- - julia -e 'using Pkg;
-             Pkg.develop("LazySets");
-             Pkg.develop("MathematicalSystems");
-             Pkg.develop("HybridSystems");
-             Pkg.develop("TaylorModels");
-             Pkg.clone(pwd());
-             Pkg.build("Reachability");
-             Pkg.test("Reachability"; coverage=true)'
+ - julia -e '$TEST_SUITE'
```
Comment (schillic) [inline .travis.yml]:
```suggestion
 - julia -e "$TEST_SUITE"
```
another try :)

**Comment by mforets** [comment]
OK let's see.. i wanted to try:

```julia
matrix:
  include:
    - julia: 1.0
      env: TO_TEST=DEVELOP
    - julia: 1.0
      env: TO_TEST=STABLE
  fast_finish: true

script:
  - if [ "$TO_TEST" = "DEVELOP" ]; then julia -e 'using Pkg; Pkg.develop("LazySets"); Pkg.develop("MathematicalSystems"); Pkg.develop("HybridSystems"); Pkg.develop("TaylorModels"); Pkg.clone(pwd()); Pkg.build("Reachability"); Pkg.test("Reachability"; coverage=true)'; fi
  - if [ "$TO_TEST" = "STABLE" ]; then julia -e 'using Pkg; Pkg.clone(pwd()); Pkg.build("Reachability"); Pkg.test("Reachability"; coverage=true)'; fi
```

**Comment by mforets** [comment]
seems to work now.. but requires https://github.com/JuliaLang/METADATA.jl/pull/22479

**Comment by schillic** [comment]
This will take a while because there was a comment, so it will have to be merged manually.

**Comment by schillic** [comment]
I think last commit
- removed the tests on Julia v1.0
- added a useless `allow_failures` because this version does not occur under `include`.

If we do not test on v1.0, we should change our `REQUIRE` file. The second item is confusing.

**Pull request merged**